File: telangana_map.c
#include <stdio.h>
#include <stdbool.h>

char *districts[] = {"Adilabad","Nirmal","Nizamabad","Kamareddy","Karimnagar","Warangal"};
char *colors[] = {"Red","Green","Blue","Yellow"};

int graph[6][6] = {
    {0,1,0,0,0,0},
    {1,0,1,0,0,0},
    {0,1,0,1,0,0},
    {0,0,1,0,1,0},
    {0,0,0,1,0,1},
    {0,0,0,0,1,0}
};

int sol[6];

bool isSafe(int node, int color) {
    for(int i=0;i<6;i++) {
        if(graph[node][i] && sol[i]==color)
            return false;
    }
    return true;
}

bool solve(int node) {
    if(node==6) return true;

    for(int c=1;c<=4;c++) {
        if(isSafe(node,c)) {
            sol[node]=c;
            if(solve(node+1)) return true;
            sol[node]=0;
        }
    }
    return false;
}

int main() {
    solve(0);
    for(int i=0;i<6;i++) {
        printf("%s -> %s\n", districts[i], colors[sol[i]-1]);
    }
    return 0;
}

File: telangana_map.py
districts = ['Adilabad','Nirmal','Nizamabad','Kamareddy','Karimnagar','Warangal']

neighbors = {
    'Adilabad': ['Nirmal'],
    'Nirmal': ['Adilabad','Nizamabad'],
    'Nizamabad': ['Nirmal','Kamareddy'],
    'Kamareddy': ['Nizamabad','Karimnagar'],
    'Karimnagar': ['Kamareddy','Warangal'],
    'Warangal': ['Karimnagar']
}

colors = ['Red','Green','Blue','Yellow']

def is_valid(district, color, assignment):
    for neighbor in neighbors[district]:
        if neighbor in assignment and assignment[neighbor] == color:
            return False
    return True

def backtrack(assignment):
    if len(assignment) == len(districts):
        return assignment

    for d in districts:
        if d not in assignment:
            for color in colors:
                if is_valid(d, color, assignment):
                    assignment[d] = color
                    result = backtrack(assignment)
                    if result:
                        return result
                    del assignment[d]
            return None

solution = backtrack({})
print("Telangana Coloring:", solution)
